client/src/components/pages/Players.js

```diff
@@ -25,6 +25,59 @@ const { Content } = Layout;
 const { Panel } = Collapse;
 const { Option } = Select;
 
+let prevPlayers = null;
+function potentialTeamCounter({ flags, hasTeams, players }) {
+  const [minPotentialTeams, setMinPotentialTeams] = useState(0);
+  const [maxPotentialTeams, setMaxPotentialTeams] = useState(0);
+
+  let isCountryBasedTeamTourney = false;
+  if (!flags.has("suiji") && !flags.has("registerAsTeam")) {
+    isCountryBasedTeamTourney = hasTeams;
+  }
+
+  if (!isCountryBasedTeamTourney) return <></>;
+
+  const countPotentialTeams = () => {
+    let countryPlayerCounts = {};
+    players.forEach(player => {
+      if (countryPlayerCounts[player.country]) {
+        countryPlayerCounts[player.country] += 1;
+      }
+      else {
+        countryPlayerCounts[player.country] = 1;
+      }
+    });
+
+    let minPotentialTeams = 0;
+    let maxPotentialTeams = 0;
+
+    Object.keys(countryPlayerCounts).forEach(k => {
+      if (countryPlayerCounts[k] >= 2) {
+        minPotentialTeams += 1;
+        maxPotentialTeams += 1;
+        if (countryPlayerCounts[k] >= 7)
+          maxPotentialTeams += 1;
+      }
+    });
+
+    return {
+      min: minPotentialTeams,
+      max: maxPotentialTeams
+    }
+  }
+  // Passing player as the dep to useEffect doesn't seem to work here, so it's done manually here
```

Comment:
hm not sure why deps array isn't working for you, ideally we'd be able to do something like this
```
const {min, max} = useMemo(countPotentialTeams, [players])
```

Reply:
Not sure if it's switching to usememo or the move to a separate component, but it seems to work now

Applied in 7a09b7cd2b7d23512599b50c4c4b5aa68f3d978b

Reply:
that's not really how `useMemo` is supposed to be used, since it's supposed to return a memorized value. but it's alright, let's just get this merged already :eh: